### ISSUE TITLE
[Bug]: Invalid version string for some default PHP version naming scheme

### ISSUE THREAD

**Issue by RiverVanRain** [Elgg/Elgg#14580]
### Elgg version

5.1.4

### Source of your Elgg installation

Elgg.org website

### Describe the bug

When we require for example `"ext-curl": "*", "ext-mbstring": "*"` in the `composer.json` and there is `'8.3.3-1+0~20240216.17+debian11~1.gbp87e37b'` PHP version on the server, we get a fatal error on the Plugins page: `Invalid version string "8.3.3-1+0~20240216.17+debian11~1.gbp87e37b"`

### Steps to reproduce

1.
2.
3.
...

### PHP version

8.3.3-1+0~20240216.17+debian11~1.gbp87e37b

### PHP error log

```text
#0 /www/vendor/composer/semver/src/Semver.php(39): Composer\Semver\VersionParser->normalize()
#1 /www/vendor/elgg/elgg/engine/classes/Elgg/Plugin/Composer.php(196): Composer\Semver\Semver::satisfies()
#2 /www/vendor/elgg/elgg/engine/classes/Elgg/Plugin/Composer.php(212): Elgg\Plugin\Composer->checkConstraints()
#3 /www/vendor/elgg/elgg/engine/classes/ElggPlugin.php(1407): Elgg\Plugin\Composer->assertRequiredPhpVersion()
#4 /www/vendor/elgg/elgg/engine/classes/ElggPlugin.php(500): ElggPlugin->assertDependencies()
#5 /www/vendor/elgg/elgg/views/default/object/plugin/full.php(38): ElggPlugin->assertCanActivate()
#6 /www/vendor/elgg/elgg/engine/classes/Elgg/ViewsService.php(491): include('...')
#7 /www/vendor/elgg/elgg/engine/classes/Elgg/ViewsService.php(426): Elgg\ViewsService->renderViewFile()
#8 /www/vendor/elgg/elgg/engine/lib/views.php(178): Elgg\ViewsService->renderView()
#9 /www/vendor/elgg/elgg/views/default/object/plugin.php(24): elgg_view()
#10 /www/vendor/elgg/elgg/engine/classes/Elgg/ViewsService.php(491): include('...')
#11 /www/vendor/elgg/elgg/engine/classes/Elgg/ViewsService.php(426): Elgg\ViewsService->renderViewFile()
#12 /www/vendor/elgg/elgg/engine/lib/views.php(178): Elgg\ViewsService->renderView()
#13 /www/vendor/elgg/elgg/engine/lib/views.php(540): elgg_view()
#14 /www/vendor/elgg/elgg/engine/lib/views.php(1256): elgg_view_entity()
#15 /www/vendor/elgg/elgg/views/default/page/components/list.php(73): elgg_view_list_item()
#16 /www/vendor/elgg/elgg/engine/classes/Elgg/ViewsService.php(491): include('...')
#17 /www/vendor/elgg/elgg/engine/classes/Elgg/ViewsService.php(426): Elgg\ViewsService->renderViewFile()
#18 /www/vendor/elgg/elgg/engine/lib/views.php(178): Elgg\ViewsService->renderView()
#19 /www/vendor/elgg/elgg/engine/lib/views.php(685): elgg_view()
#20 /www/vendor/elgg/elgg/views/default/admin/plugins.php(23): elgg_view_entity_list()
#21 /www/vendor/elgg/elgg/engine/classes/Elgg/ViewsService.php(491): include('...')
#22 /www/vendor/elgg/elgg/engine/classes/Elgg/ViewsService.php(426): Elgg\ViewsService->renderViewFile()
#23 /www/vendor/elgg/elgg/engine/lib/views.php(178): Elgg\ViewsService->renderView()
#24 /www/vendor/elgg/elgg/views/default/resources/admin.php(26): elgg_view()
#25 /www/vendor/elgg/elgg/engine/classes/Elgg/ViewsService.php(491): include('...')
#26 /www/vendor/elgg/elgg/engine/classes/Elgg/ViewsService.php(426): Elgg\ViewsService->renderViewFile()
#27 /www/vendor/elgg/elgg/engine/lib/views.php(332): Elgg\ViewsService->renderView()
#28 /www/vendor/elgg/elgg/engine/classes/Elgg/Router.php(211): elgg_view_resource()
#29 /www/vendor/elgg/elgg/engine/classes/Elgg/Router.php(121): Elgg\Router->prepareResponse()
#30 /www/vendor/elgg/elgg/engine/classes/Elgg/Router.php(101): Elgg\Router->getResponse()
#31 /www/vendor/elgg/elgg/engine/classes/Elgg/Application.php(448): Elgg\Router->route()
#32 /www/vendor/elgg/elgg/engine/classes/Elgg/Application.php(340): Elgg\Application->run()
#33 /www/vendor/elgg/elgg/engine/classes/Elgg/Application.php(390): Elgg\Application::route()
#34 /www/index.php(8): Elgg\Application::index()
#35 {main}
```

**Comment by jeabakker** [comment]
this is a bug in the Composer Semver dependency, not something we can easily fix.

I reported the issue  https://github.com/composer/semver/issues/157

**Comment by jeabakker** [comment]
For now Composer isn't willing to fix the issue. We catch the exception in order to prevent the page from crashing.

Since there is an exception we will fail the version check. this could prevent you from enabling a plugin. A workaround would be to temporarily disable/remove the conflicting requirement.

**Comment by jeabakker** [comment]
fixed in #14576

**Comment by RiverVanRain** [comment]
Got it. Thanks
